crates/router/src/middleware.rs

```diff
@@ -81,3 +84,114 @@ pub fn default_response_headers() -> actix_web::middleware::DefaultHeaders {
+                }
+            }
+            result
+        }
+        serde_json::Value::Object(obj) => {
+            let mut result = String::new();
+            for (index, (key, value)) in obj.iter().enumerate() {
+                let child_key = format!("{}[{}]", parent_key, key);
+                result.push_str(&get_request_details_from_value(value, &child_key));
+                if index < obj.len() - 1 {
+                    result.push_str(", ");
+                }
+            }
+            result
+        }
+    }
+}
+
+/// Middleware for Logging request_details of HTTP 400 Bad Requests
+pub struct Http400RequestDetailsLogger;
+
+impl<S: 'static, B> actix_web::dev::Transform<S, actix_web::dev::ServiceRequest>
+    for Http400RequestDetailsLogger
+where
+    S: actix_web::dev::Service<
+        actix_web::dev::ServiceRequest,
+        Response = actix_web::dev::ServiceResponse<B>,
+        Error = actix_web::Error,
+    >,
+    S::Future: 'static,
+    B: 'static,
+{
+    type Response = actix_web::dev::ServiceResponse<B>;
+    type Error = actix_web::Error;
+    type Transform = Http400RequestDetailsLoggerMiddleware<S>;
+    type InitError = ();
+    type Future = std::future::Ready<Result<Self::Transform, Self::InitError>>;
+
+    fn new_transform(&self, service: S) -> Self::Future {
+        std::future::ready(Ok(Http400RequestDetailsLoggerMiddleware {
+            service: std::rc::Rc::new(service),
+        }))
+    }
+}
+
+pub struct Http400RequestDetailsLoggerMiddleware<S> {
+    service: std::rc::Rc<S>,
+}
+
+impl<S, B> actix_web::dev::Service<actix_web::dev::ServiceRequest>
+    for Http400RequestDetailsLoggerMiddleware<S>
+where
+    S: actix_web::dev::Service<
+            actix_web::dev::ServiceRequest,
+            Response = actix_web::dev::ServiceResponse<B>,
+            Error = actix_web::Error,
+        > + 'static,
+    S::Future: 'static,
+    B: 'static,
+{
+    type Response = actix_web::dev::ServiceResponse<B>;
+    type Error = actix_web::Error;
+    type Future = futures::future::LocalBoxFuture<'static, Result<Self::Response, Self::Error>>;
+
+    actix_web::dev::forward_ready!(service);
+
+    fn call(&self, mut req: actix_web::dev::ServiceRequest) -> Self::Future {
+        let svc = self.service.clone();
+        let request_id_fut = req.extract::<router_env::tracing_actix_web::RequestId>();
+        Box::pin(async move {
+            let (http_req, payload) = req.into_parts();
+            let result_payload: Vec<Result<bytes::Bytes, actix_web::error::PayloadError>> =
+                payload.collect().await;
+            let payload = result_payload
+                .into_iter()
+                .collect::<Result<Vec<bytes::Bytes>, actix_web::error::PayloadError>>()?;
+            let bytes = payload.clone().concat().to_vec();
+            let value: serde_json::Value = serde_json::from_slice(&bytes)?;
+            let concatenated_bytes: bytes::Bytes = payload.concat().to_vec().into();
```

Comment:
Can we create payload from the `bytes` from line number 177? And move this logic inside the if condition so that we perform this operation only in case of 400 status code